src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/Marshal/PInvokeErrorMessageTests.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class PInvokeErrorMessageTests
+    {
+        public static IEnumerable<object[]> GetErrorCode_TestData()
+        {
+            yield return new object[] { 0 };
+            yield return new object[] { 1 };
+
+            // errno values
+            yield return new object[] { 0x10002 };
+            yield return new object[] { 0x10003 };
+            yield return new object[] { 0x10014 };
+            yield return new object[] { 0x1001D };
+
+            // HRESULT values
+            yield return new object[] { unchecked((int)0x80004001) };
+            yield return new object[] { unchecked((int)0x80004005) };
+            yield return new object[] { unchecked((int)0x80070057) };
+            yield return new object[] { unchecked((int)0x8000FFFF) };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetErrorCode_TestData))]
+        public void PInvokeErrorMessage_Returns_Win32Exception_Message(int error)
+        {
+            // The Win32Exception represents the canonical system exception on
```

Comment:
I wouldn't say it represents the canonical system exception, especially now that you don't need it to get this message.

It's actually not thrown in many places outside of Windows. Essentially Process and Networking, and many of those could throw an existing derived type, then we could squint and pretend it doesn't exist except on Windows.
https://github.com/dotnet/runtime/issues/26227#issuecomment-747842770

It doesn't matter of course, just taking exception to the comment..